NuRadioMC/utilities/medium.py

```diff
@@ -1,112 +1,188 @@
+            delta_n = 0.423)
 
 
-class ARAsim_southpole(medium):
-
+class ARAsim_southpole(IceModel_Simple):
     def __init__(self):
         # define model parameters (SPICE 2015/southpole)
-        self.n_ice = 1.78
-        self.z_0 = 75.75757575757576 * units.m
-        self.delta_n = 0.43
-
+        super().__init__(
+            z_bottom = -2820*units.meter, 
+            n_ice = 1.78, 
+            z_0 = 75.75757575757576*units.meter, 
+            delta_n = 0.43)
 
-class mooresbay_simple(medium):
 
-    # from https://doi.org/10.1088/1475-7516/2018/07/055 MB1 model
+class mooresbay_simple(IceModel_Simple):
+    def __init__(self):
+        # from https://doi.org/10.1088/1475-7516/2018/07/055 MB1 model
+        super().__init__(
+            n_ice = 1.78, 
+            z_0 = 34.5*units.meter, 
+            delta_n = 0.46)
+
+        # from https://doi.org/10.3189/2015JoG14J214
+        self.add_reflective_bottom( 
+            refl_z = -576*units.m, 
+            refl_coef = 0.82, 
+            refl_phase_shift = 180*units.deg)
+
+
+class mooresbay_simple_2(IceModel_Simple):
+    def __init__(self):\
+        # from https://doi.org/10.1088/1475-7516/2018/07/055 MB2 model
+        super().__init__(
+            n_ice = 1.78, 
+            z_0 = 37*units.meter, 
+            delta_n = 0.481)
+
+        # from https://doi.org/10.3189/2015JoG14J214
+        self.add_reflective_bottom( 
+            refl_z = -576*units.m, 
+            refl_coef = 0.82, 
+            refl_phase_shift = 180*units.deg)
+
+
+class greenland_simple(IceModel_Simple):
     def __init__(self):
-        self.n_ice = 1.78
-        self.z_0 = 34.5 * units.m
-        self.delta_n = 0.46
-        self.reflection = -576 * units.m  # from https://doi.org/10.3189/2015JoG14J214
-        self.reflection_coefficient = 0.82  # from https://doi.org/10.3189/2015JoG14J214
-        self.reflection_phase_shift = 180 * units.deg
+        # from C. Deaconu, fit to data from Hawley '08, Alley '88
+        # rho(z) = 917 - 602 * exp (-z/37.25), using n = 1 + 0.78 rho(z)/rho_0
+        super().__init__(
+            z_bottom = -3000*units.meter, 
+            n_ice = 1.78, 
+            z_0 = 37.25*units.meter, 
+            delta_n = 0.51)
+
+class greenland_firn(IceModel):
+    """
+    This model can only be used with the radiopropa raytracer.
+    Therefor, the model is implemented through radiopropa.
+    """
+    def __init__(self):
+        if not radiopropa_is_imported:
+            logger.error('This ice model depends fully on RadioPropa, which was not import, and can therefore not be used. \nMore info on https://github.com/nu-radio/RadioPropa')
+            raise ImportError('This ice model depends fully on RadioPropa, which could not be imported')
+
+        super().__init__(z_bottom = -3000*units.meter)
+        self.z_firn = -14.9*units.meter
+        
+        self._scalarfield = RP.IceModel_Firn(
+            z_surface = self.z_airBoundary*RP.meter/units.meter,
```

Comment:
not so obvious why in the ice model we define in RP units? and RP.meter/units.meter? I am sure you checked this?

Reply:
the thing is that radiopropa and NuRadioMC use both their own unit system, so when running radiopropa modules I convert from the NuRadioMC unit system to the RP unit system and then back again as the RP module spits out its calculation. I did this to be sure everything gets handled right in the RP module. Not sure if this could be done in a better way.

Reply:
ok fine

Reply:
The philosophy is that any number used in NuRadioReco/MC should be in the nuRadio unit system. So the best way would be to convert units whenever values are passed to RadioPropa

Reply:
which should be the case at the moment